be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -67,8 +67,13 @@ BetaRowsetWriter::~BetaRowsetWriter() {
 Status BetaRowsetWriter::init(const RowsetWriterContext& rowset_writer_context) {
     _context = rowset_writer_context;
     _rowset_meta.reset(new RowsetMeta);
-    if (_context.data_dir) {
+    if (_context.fs == nullptr && _context.data_dir) {
         _rowset_meta->set_fs(_context.data_dir->fs());
+    } else {
+        _rowset_meta->set_fs(_context.fs);
+    }
+    if (_context.resource_id.size() > 0) {
+        _rowset_meta->set_resource_id(_context.resource_id);
```

Comment:
why rowset meta need save resource_id? I did not find use of it.

Reply:
it is used when calling function is_local()